### ISSUE TITLE
Feature Request: Configuration to define exactly what fields would be updated in a insert with conflict: update

### ISSUE THREAD

**Issue by thelinuxlich** [rethinkdb/rethinkdb#3989]
An example of use case would be a table with created_at and updated_at fields, if the record exists, update only the updated_at field.

**Comment by thelinuxlich** [comment]
Is this operation atomic?

```
r.table("sessions").get(ID).replace(function(row) {
   return r.branch(row.eq(null), 
       { id: ID,created_at: r.now(), updated_at: r.now() }, 
       row.merge({updated_at: r.now() }
   );
});
```

**Comment by thelinuxlich** [comment]
confirmed with @neumino that this operation is atomic, so all is good. I think you should add an entry for this operation on the cookbook, it's so common :)

**Comment by danielmewes** [comment]
Good idea @thelinuxlich . Opened https://github.com/rethinkdb/docs/issues/694

**Comment by deontologician** [comment]
This is also something that could be solved by #3753
